app/components/submission/ManuscriptMetadata/SubjectAreaDropdown.test.js

```diff
@@ -0,0 +1,120 @@
+    }
+
+    it('focusing causes the dropdown to open & display every option', () => {
+      typeSubjectArea('') // typing nothing into the field seems to be the only way to programmatically focus react-select)
+      expect(wrapper.find('Menu').exists()).toEqual(true)
+      expect(wrapper.find('Option')).toHaveLength(17)
+    })
+
+    it('an option can be selected from the dropdown', () => {
+      typeSubjectArea('b')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      expect(onChange.mock.calls).toHaveLength(1)
+      expect(wrapper.find('Menu').exists()).toEqual(false)
+      expect(wrapper.find('MultiValue')).toHaveLength(1) // MultiValue = a "tag" (i.e. selected option) in react-select
+      expect(
+        wrapper
+          .find('MultiValue')
+          .contains('Biochemistry and Chemical Biology'),
+      ).toBe(true)
+    })
+
+    it('2 options can be selected from the dropdown', () => {
+      typeSubjectArea('b')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      typeSubjectArea('c')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      expect(wrapper.find('MultiValue')).toHaveLength(2)
+      expect(wrapper.find('MultiValue').contains('Cancer Biology')).toBe(true)
+
+      // selectWrapper.props().onInputChange('E')
+      // selectInput.simulate('change')
+      // expect(wrapper.contains('Menu')).toBe(false)
+    })
+
+    it('selecting 2 options disables further selection', () => {
+      typeSubjectArea('b')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      typeSubjectArea('c')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      expect(wrapper.contains('DropdownIndicator')).toBe(false)
+    })
+
+    it('selection/deselection of 2 items triggers success message to show/hide', () => {
+      typeSubjectArea('b')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      typeSubjectArea('c')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      expect(wrapper.contains('No more than two subject areas')).toBe(true)
+
+      const crossOnFirstTag = wrapper.find('MultiValueRemove').at(0)
+      crossOnFirstTag.simulate('click')
+      expect(wrapper.find('MultiValue')).toHaveLength(1)
+      expect(wrapper.contains('No more than two subject areas')).toBe(false)
+    })
+
+    it('typing filters list of options based on first letter(s)', () => {
+      typeSubjectArea('c')
+      const totalStartingWithC = wrapper.find('Option').length
+      expect(totalStartingWithC).toBeLessThan(17)
+      wrapper.find('Option').forEach(option => {
+        const optionText = option.childAt(0).text()
+        const wordStartsWithC = new RegExp(/^[cC]\w+/)
+        expect(optionText).toMatch(wordStartsWithC)
+      })
+
+      typeSubjectArea('ce')
+      const totalStartingWithCe = wrapper.find('Option').length
+      expect(totalStartingWithCe).toBeLessThan(totalStartingWithC)
```

Comment:
Again, what if all options starting with `c` will change to option starting with `ce`?

Reply:
If I'm understanding what you're asking correctly, the `totalStartingWithCe` variable doesn't update, when the number of Options in the "HTML" changes. So there's no danger of the "change" you're referring to.

It's a bit annoying actually, to repeat yourself. But if you want to find the new number of `Option`s on the page, you have to call `wrapper.find` again.

Does that answer your question? I'm not sure if I understood you.

Reply:
No, I meant what if eLife will change the number of options starting with `c` and `ce` in the future so they are equal? Is this a relevant test? I wouldn't think so.

Reply:
See comment above
> But you're totally right. The toBeLessThan test is actually totally redundant, since I'm checking whether they start with the appropriate letters anyway. Thanks +1